app/backend/approaches/chatreadretrieveread.py

```diff
@@ -249,6 +249,20 @@ async def run_with_streaming(
             if event["choices"]:
                 yield event
 
+    async def run(
+        self, messages: list[dict], stream: bool = False, session_state: Any = None, context: dict[str, Any] = {}
+    ) -> Union[dict[str, Any], AsyncGenerator[dict[str, Any], None]]:
+        overrides = context.get("overrides", {})
+        auth_claims = context.get("auth_claims", {})
+        if stream is False:
+            # Workaround for: https://github.com/openai/openai-python/issues/371
+            async with aiohttp.ClientSession() as s:
+                openai.aiosession.set(s)
+                response = await self.run_without_streaming(messages, overrides, auth_claims)
```

Comment:
In the future if we add anything to `context`, we can refactor the `run_without_streaming` and `run_with_streaming` methods to take the context object instead of unpacking it into overrides and auth claims. But this is good for now